okapi-core/src/main/java/org/folio/okapi/util/TokenHeader.java

```diff
@@ -0,0 +1,73 @@
+package org.folio.okapi.util;
+
+import io.vertx.core.MultiMap;
+import org.folio.okapi.common.XOkapiHeaders;
+
+public class TokenHeader {
+
+  private TokenHeader() { }
+
+  /**
+   * Get cookie value from name.
+   *
+   * <p>Based on RFC6265.
+   * <a href="https://www.rfc-editor.org/rfc/rfc6265#section-4.2.1">Cookie syntax</a>
+   *    <a href="https://www.rfc-editor.org/rfc/rfc6265#section-4.1.1">Cookie pair syntax</a>
+   * @param cookie cookie string
+   * @param name name to look for
+   * @return value; null if not found
+   * @throws IllegalArgumentException if name occurs multiple times.
+   */
+  static String cookieValue(String cookie, String name) {
+    String[] components = cookie.trim().split("; ");
```

Comment:
Using existing parser `ServerCookieDecoder.STRICT.decodeAll(cookie)` avoids extensive tests of edge cases.

Reply:
Ok. Will look into that